src/Tasks.UnitTests/XmlPoke_Tests.cs

```diff
@@ -174,6 +169,23 @@ public void PokeMissingParams()
             }
         }
 
+        [Fact]
+        // https://github.com/dotnet/msbuild/issues/5814
+        public void XmlPokeWithEmptyValue()
+        {
+            string xmlInputPath;
+            Prepare(_xmlFileNoNs, out xmlInputPath);
+            string projectContents = @"
```

Comment:
No need for change here - just an information. From C# 11 you can use [raw string literals](https://learn.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-11#raw-string-literals) - sou you wouldn't need to combine quotes here.

```C#
string projectContents = """
<Project ToolsVersion="msbuilddefaulttoolsversion">
                <Target Name="Poke">
                    <XmlPoke Value="" Query="//class/variable/@Name" XmlInputPath="{0}"/>
                </Target>
                </Project>""";
```